### PR TITLE
enable partially entangling ms gates

### PR DESCRIPTION
**Pull request by splch**
adds a third parameter for the ms gate that defaults to `0.25` (fully entangling). can be changed to enable partial entanglement.

### CODE REVIEW

File: pennylane_ionq/ops.py [hunk 1]
```diff
@@ -61,28 +61,34 @@ class GPI2(Operation):  # pylint: disable=too-few-public-methods
...
+                0 & -i e^{2 \pi i(\phi_{0}-\phi_{1})} & \cos(\theta / 2) & 0 \\
+                -i e^{2 \pi i(\phi_{0}+\phi_{1})} & 0 & 0 & \cos(\theta / 2)
             \end{pmatrix}
+
     Args:
-        phi0 (float): phase of the first qubit :math:`\phi`
-        phi1 (float): phase of the second qubit :math:`\phi`
+        phi0 (float): phase of the first qubit :math:`\phi_0`
+        phi1 (float): phase of the second qubit :math:`\phi_1`
+        theta (float): entanglement ratio of the qubits :math:`\theta` [0, 0.25], defaults to 0.25
```
Comment (trbromley) [inline pennylane_ionq/ops.py:80]:
```suggestion
        theta (float): entanglement ratio of the qubits :math:`\theta \in [0, 0.25]`, defaults to 0.25
```

**Review by trbromley — APPROVED** [review]
Thanks a lot @splch! I'll tag someone from the team to take a second look.

**Comment by splch** [comment]
thanks! i appreciate the suggestion too :)

File: pennylane_ionq/ops.py [hunk 2]
```diff
@@ -61,28 +61,34 @@ class GPI2(Operation):  # pylint: disable=too-few-public-methods
...
+        phi0 (float): phase of the first qubit :math:`\phi_0`
+        phi1 (float): phase of the second qubit :math:`\phi_1`
+        theta (float): entanglement ratio of the qubits :math:`\theta \in [0, 0.25]`, defaults to 0.25
         wires (Sequence[int]): the subsystems the operation acts on
     """
 
-    num_params = 2
+    num_params = 3
     num_wires = 2
+    par_domain = "R"
```
Comment (albi3ro) [inline pennylane_ionq/ops.py:86]: I think this was removed quite some time ago.   Does anything in the ionq plugin need it?

```suggestion
```

**Review by albi3ro — COMMENTED** [review]
Looks straightforward enough.  

Mind adding a change log entry at:

https://github.com/PennyLaneAI/PennyLane-IonQ/blob/master/CHANGELOG.md

File: pennylane_ionq/ops.py (hunk 2 above)
Comment (splch) [reply pennylane_ionq/ops.py:86]: no nothing needs this - i saw in it in some docs but will remove :)

**Comment by trbromley** [comment]
Thanks @splch! Looks like it's nearly there, but missing a test that passes the `theta` parameter. Are you able to add that in?

**Comment by splch** [comment]
> Thanks @splch! Looks like it's nearly there, but missing a test that passes the `theta` parameter. Are you able to add that in?

hi! thanks for the feedback :) is this what you were thinking of for theta testing? https://github.com/PennyLaneAI/PennyLane-IonQ/pull/101/files#diff-72107d6f4a703abaeb6a7cf709e0c99f7ed0b7c74f0b14b0cbc4f35c7c650e26R369-R389

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,12 +12,16 @@
 
 ### Documentation 📝
 
+* Updated the docstring for the MS operation to include the new `theta` parameter, explaining its use and effects on the gate operation.
```
Comment (astralcai) [inline CHANGELOG.md:15]: The wording sounded like it was documentation added for an existing feature, but it is actually a new feature, so probably change it to:
```suggestion
* The MS operation now includes a new `theta` parameter to specify the entanglement ratio of the qubits.
```

**Review by astralcai — APPROVED** [review]

**Comment by Alex-Preciado** [comment]
Thank you so much for this contribution @splch !! We have double checked and the Codecov failures seem unrelated to the changes made in this PR. Force merging now!! 🚀

**Pull request merged**